src/plotting.jl

```diff
@@ -7,19 +7,20 @@
     principalcoord(pc, 1), principalcoord(pc,2)
 end
 
-@recipe function f(abun::AbundanceTable; topabund::Int=10, sorton::Symbol=:top)
-    in(sorton, [:top, :hclust, abun.samples...]) || error("invalid sorton option")
+@userplot AbundancePlot
+@recipe function f(hb::AbundancePlot; topabund=10, sorton=:top)
+    abun = hb.args[1]
+
+    topabund = min(topabund, nspecies(abun))
+    in(sorton, [:top, :hclust, Symbol.(sitenames(abun))...]) || error("invalid sorton option") #replace `, abun.samples...` in the Array, but the code only handles :top and :hclust below anyway
     2 < topabund < 12 || error("n must be between 2 and 12")
 
     top = filterabund(abun, topabund)
 
-    c = distinguishable_colors(topabund+1)
```

Comment:
Not sure why you'd need this?

Reply:
Don't remember... a lot of this plotting code is a real hackjob

Reply:
I think your unnecessary `@series` call forced all subsequent series to have the same color.